lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -696,6 +697,63 @@ namespace Js
         return builder.ToString();
     }
 
+
+    // Relative indexing proposal 
+    // String.prototype.at(index): https://tc39.es/proposal-relative-indexing-method/#sec-string.prototype.at
+    // Spec: https://tc39.es/proposal-relative-indexing-method
+    // Github: https://github.com/tc39/proposal-relative-indexing-method
+    Var JavascriptString::EntryAt(RecyclableObject* function, CallInfo callInfo, ...) {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+        JS_REENTRANCY_LOCK(jsReentLock, scriptContext->GetThreadContext());
+
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        JavascriptString * pThis = nullptr;
+
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        // 2. Let S be ? ToString(O).
+        JS_REENTRANT(jsReentLock, GetThisStringArgument(args, scriptContext, _u("String.prototype.at"), &pThis));
+
+        // 3. Let len be the length of S.
+        charcount_t len = pThis->GetLength();
+
+        // 4. Let relativeIndex be ? ToInteger(index).
+        int64_t relativeIndex = 0;
+
+        if (args.Info.Count > 1)
+        {
+            JS_REENTRANT(jsReentLock, relativeIndex = NumberUtilities::TryToInt64(JavascriptConversion::ToInteger(args[1], scriptContext)));
+        }
+
+        // 5. If relativeIndex >= 0, then
+        //     a. Let k be relativeIndex.
+        // 6. Else,
+        //     a. Let k be len + relativeIndex.
+        int64_t k = relativeIndex;
+        
+        if (relativeIndex < 0)
+        {
+            k += len;
+        }
+        
+        // 7. If k < 0 or k >= len, then return undefined.
+        if (k < 0 || k >= (int64_t)len)
+        {
+            return scriptContext->GetLibrary()->GetUndefined();
+        }
+        
+        Var value;
+        Assert(pThis->GetItemAt((charcount_t)k, &value));
```

Comment:
Asserts are compiled out on release builds so need to do the Get on the line before the Assert.

Reply:
Is change from Assert to AssertOrFailFast ok?

Reply:
The test failure just before you pushed the latest update appeared to be hitting the FailFast you'd added - be good to work out what case could trigger this condition if it's not meant to happen.

Reply:
So MSVC doesn't like the possibility of the Var value being undefined.

Looking at it GetItemAt should only leave you undefined if k is > length - which you've already checked above.

So I'd just put this:
```c
Var value = nullptr;
pThis->GetItemAt((charcount_t)k, &value);
Assert(value != nullptr);
```

Reply:
i'll put that if revert of this if will fail.